endpoints.tf

```diff
@@ -13,3 +13,147 @@ resource "aws_vpc_endpoint" "transfer_server" {
+}
+
+data "aws_vpc_endpoint_service" "ec2_endpoint" {
+  count   = length(var.ec2_endpoint.subnet_ids) != 0 ? 1 : 0
+  service = "ec2"
+}
+
+resource "aws_security_group" "sgp_ec2_endpoint" {
+  count  = length(var.ec2_endpoint.security_group_ids) == 0 ? 1 : 0
+  vpc_id = aws_vpc.default.id
+  name   = "sgp-ec2-endpoint-${var.name}"
+  tags   = { Name = "sgp-ec2-endpoint-${var.name}" }
+}
+
+resource "aws_security_group_rule" "sgr_ec2_endpoint" {
+  count             = length(var.ec2_endpoint.security_group_ids) == 0 ? 1 : 0
+  type              = "ingress"
+  from_port         = 443
+  to_port           = 443
+  protocol          = "tcp"
+  cidr_blocks       = [aws_vpc.default.cidr_block]
+  security_group_id = aws_security_group.sgp_ec2_endpoint[0].id
+}
+
+resource "aws_vpc_endpoint" "ec2_endpoint" {
+  count               = length(var.ec2_endpoint.subnet_ids) != 0 ? 1 : 0
+  private_dns_enabled = var.ec2_endpoint.private_dns_enabled
+  security_group_ids  = length(var.ec2_endpoint.security_group_ids) == 0 ? [aws_security_group.sgp_ec2_endpoint[count.index].id] : var.ec2_endpoint.security_group_ids
+  service_name        = data.aws_vpc_endpoint_service.ec2_endpoint[0].service_name
+  subnet_ids          = var.ec2_endpoint.subnet_ids
+  tags                = merge(var.tags, { "Name" = "ec2-endpoint-${var.name}" })
+  vpc_endpoint_type   = "Interface"
+  vpc_id              = aws_vpc.default.id
+}
+
+data "aws_vpc_endpoint_service" "ssmmessages_endpoint" {
+  count   = length(var.ssmmessages_endpoint.subnet_ids) != 0 ? 1 : 0
+  service = "ssmmessages"
+}
+
+resource "aws_security_group" "sgp_ssmmessages_endpoint" {
+  count  = length(var.ssmmessages_endpoint.security_group_ids) == 0 ? 1 : 0
+  vpc_id = aws_vpc.default.id
+  name   = "sgp-ssmmessages-endpoint-${var.name}"
+  tags   = { Name = "sgp-ssmmessages-endpoint-${var.name}" }
+}
+
+resource "aws_security_group_rule" "sgr_ssmmessages_endpoint" {
+  count             = length(var.ssmmessages_endpoint.security_group_ids) == 0 ? 1 : 0
+  type              = "ingress"
+  from_port         = 443
+  to_port           = 443
+  protocol          = "tcp"
+  cidr_blocks       = [aws_vpc.default.cidr_block]
+  security_group_id = aws_security_group.sgp_ssmmessages_endpoint[0].id
+}
+
+resource "aws_vpc_endpoint" "ssmmessages_endpoint" {
+  count               = length(var.ssmmessages_endpoint.subnet_ids) != 0 ? 1 : 0
+  private_dns_enabled = var.ssmmessages_endpoint.private_dns_enabled
+  security_group_ids  = length(var.ssmmessages_endpoint.security_group_ids) == 0 ? [aws_security_group.sgp_ssmmessages_endpoint[count.index].id] : var.ssmmessages_endpoint.security_group_ids
+  service_name        = data.aws_vpc_endpoint_service.ssmmessages_endpoint[0].service_name
+  subnet_ids          = var.ssmmessages_endpoint.subnet_ids
+  tags                = merge(var.tags, { "Name" = "ssmmessages-endpoint-${var.name}" })
+  vpc_endpoint_type   = "Interface"
+  vpc_id              = aws_vpc.default.id
+}
```

Comment:
A few changes needed here, please consider the following:

- Sort each resource's fields as per comment in `examples/all/main.tf` diff
- Avoid the resource type in the resource name, e.g.
  - `resource "aws_security_group" "sgp_ssmmessages_endpoint" ` vs `resource "aws_security_group" "ssmmessages_endpoint"`
  - `resource "aws_vpc_endpoint" "ssmmessages_endpoint"` vs `resource "aws_vpc_endpoint" "ssmmessages"`
- Consider separating each VPC endpoint with a comment for easier readability